src/components/apiExamples/apiExamples.tsx

```diff
@@ -25,29 +29,39 @@ function Example({api, selectedTabView, selectedResponse}: ExampleProps) {
     exampleJson = api.responses[selectedResponse].content?.example;
   }
 
+  // load prism dynamically for these codeblocks,
+  // otherwise the highlighting applies globally
+  useEffect(() => {
+    (async () => {
+      const {highlightAllUnder} = await import('prismjs');
+      await import('prismjs/components/prism-json');
+      if (ref.current) {
+        highlightAllUnder(ref.current);
+      }
+    })();
+  }, [selectedResponse, selectedTabView]);
+
   return (
-    <pre className="api-block-example response">
+    <pre className={responseStyles} ref={ref}>
       {selectedTabView === 0 &&
         (exampleJson ? (
           <code
+            className={jsonCodeBlockStyles}
             dangerouslySetInnerHTML={{
-              __html: Prism.highlight(
-                JSON.stringify(exampleJson, null, 2),
-                Prism.languages.json,
-                'json'
-              ),
+              __html: JSON.stringify(exampleJson, null, 2),
             }}
           />
         ) : (
           strFormat(api.responses[selectedResponse].description)
         ))}
       {selectedTabView === 1 && (
         <code
+          className={jsonCodeBlockStyles}
           dangerouslySetInnerHTML={{
-            __html: Prism.highlight(
-              JSON.stringify(api.responses[selectedResponse].content?.schema, null, 2),
-              Prism.languages.json,
-              'json'
+            __html: JSON.stringify(
+              api.responses[selectedResponse].content?.schema,
+              null,
+              2
```

Comment:
Is there any way uncontrolled input can land here?

Reply:
good point, but no, this is just the filtered response of our openAPI call